### PR TITLE
[autograd] Fix logcumsumexp JVP dropping early-prefix contributions

### LINKED ISSUE #196705: torch.logcumsumexp loses a first-order JVP contribution
### 🐛 Describe the bug

`torch.logcumsumexp` returns `-5.9999999999994005` for a first-order JVP whose mathematically correct value is `-5.0` in `float64`.
Every prefix is a smooth log-sum-exp of finite logits. The one-element prefix must keep the derivative of its input. At `t = -400.0`, summing the explicit prefix log-sum-exp expressions gives JVP `-5.0`. PyTorch returns about `-6.0`, as if that length-1 contribution were dropped. The forward value matches.
This issue reproduces on CPU, so CUDA availability does not affect the result.
```python
import torch
dtype = torch.float64
def f(t):
    x = torch.stack((t, 1 - t, 2 * t + 3, -t, t + 2, 3 * t - 1)).reshape(2, 3)
    w = torch.tensor([[1, 2, -1], [3, -2, 4]], dtype=t.dtype)
    return (torch.logcumsumexp(x, dim=-1) * w).sum()
t = torch.tensor(-400.0, dtype=dtype)
_, actual = torch.func.jvp(f, (t,), (torch.ones_like(t),))
expected = torch.tensor(-5.0, dtype=dtype)
print("actual:")
print(actual)
print("expected:")
print(expected)
```
### Actual vs expected result
```text
actual:
-5.9999999999994005
expected:
-5.0
```
The discrepancy occurs in the first-order forward-mode derivative. The forward value is `2001.0` and matches the prefix log-sum-exp; the JVP is off by about `1`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A
OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64
Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01
CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128
Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @albanD

### PR DESCRIPTION
**Pull request by indrajeetapache**
## Summary

Fixes #196705.

`torch.func.jvp` through `torch.logcumsumexp` returns about `-6` instead of `-5` on the issue repro (`float64`, `t = -400`). Forward values match; only the first-order JVP is wrong.

**Root cause:** `logcumsumexp_jvp` in `FunctionsManual.cpp` copied `logsumexp_jvp`'s **global** max-along-dim exp-normalize trick, then did `cumsum(exp(x-M)*t) / cumsum(exp(x-M))`. That is correct for a full reduction, not for prefixes. When an early prefix is far below a later max, `exp(x_early - M)` underflows to 0, so the length-1 JVP becomes 0 instead of the input tangent (off-by-about-1 in the weighted sum).

**Fix:** use the recurrence that follows from `y[i] = logaddexp(y[i-1], x[i])`:

```text
y_t[i] = exp(y[i-1] - y[i]) * y_t[i-1] + exp(x[i] - y[i]) * x_t[i]
```

with `y = logcumsumexp(x)`. Early prefixes stay well-scaled relative to their own running log-sum-exp.

## Test plan

- [x] `test/test_ops_fwd_gradients.py::TestFwdGradients::test_logcumsumexp_jvp_large_range` (issue repro)
- [ ] CI: forward-AD / autograd CPU tests

cc @albanD @soulitzer @lezcano

### CODE REVIEW

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -1192,36 +1192,38 @@ Tensor logcumsumexp_backward(
...
 
-  constexpr double eps = 1e-13;
+  auto scalar_min = AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+      at::ScalarType::Half,
+      at::ScalarType::BFloat16,
+      at::typeMetaToScalarType(self_t.dtype()),
+      "logcumsumexp_jvp",
+      []() { return c10::Scalar(std::numeric_limits<scalar_t>::lowest()); });
 
-  if (areAnyTensorSubclassLike({self_p, self_t})) {
```
Comment (Skylion007) [inline torch/csrc/autograd/FunctionsManual.cpp:1217]: We still the areAnyTensorSubclassLike to use the in place op dispatch

**Comment by Skylion007** [comment]
@indrajeetapache please remove the xfails

**Comment by Skylion007** [comment]
@claude show what xfails need to be dropped

**Comment by indrajeetapache** [comment]
Addressed the CI / review notes:

1. Removed the `logcumsumexp` `complex128` xfail from `test_forward_mode_AD` only (the unexpected-success failures). Left the `test_fn_fwgrad_bwgrad` xfail in place.
2. Deleted the stale OpInfo comment about `max_values_cpu` / ComplexDouble; kept the inductor `toleranceOverride`.

@Skylion007 on `areAnyTensorSubclassLike`: the current JVP path is all out-of-place (`where` / `logcumsumexp` / `exp` / subtract) and does not mutate inputs with inplace ops, so that branch is not needed here. Happy to adjust if you still want an explicit subclass path.
